indexer.go

```diff
@@ -0,0 +1,31 @@
+package indexer
+
+// Provider interface of needed provider functions
+type Provider interface {
+	GetBlockHeight() (int, error)
+	// TODO: add missing functions
+}
+
+// Database interface of needed database functions
+type Database interface {
+	// TODO: add missing functions
+}
+
+// Indexer struc handler for Indexer functions
+type Indexer struct {
+	provider Provider
+	database Database
+}
+
+// NewIndexer returns Indexer instance with given input
+func NewIndexer(provider Provider, database Database) *Indexer {
```

Comment:
database is locking the users to think it must explicitly be a database, I think a `writer` interface fits better the purpose